frontend/src/actions/transfers/subsidized-transfer.ts

```diff
@@ -0,0 +1,47 @@
+import type { IEthereumProvider } from '@/services/web3-provider';
+import { UInt256 } from '@/types/uint-256';
+
+import { type TransferData, Transfer } from './transfer';
+
+export class SubsidizedTransfer extends Transfer {
+  readonly feeSubAddress: string;
+  readonly originalRequestManagerAddress: string;
+
+  constructor(data: SubsidizedTransferData) {
+    if (!data.sourceChain.feeSubAddress) {
+      throw new Error('Please provide a fee subsidy contract address.');
+    }
+    super(data);
+    this.originalRequestManagerAddress = data.sourceChain.requestManagerAddress;
+    this.feeSubAddress = data.sourceChain.feeSubAddress;
+    Object.assign(this.fees.uint256, this.fees.uint256.multiply(new UInt256(0)));
+  }
+
+  async ensureTokenAllowance(provider: IEthereumProvider): Promise<void> {
+    if (this.feeSubAddress) {
+      this.sourceChain.requestManagerAddress = this.feeSubAddress;
+    }
+    await super.ensureTokenAllowance(provider);
+    this.sourceChain.requestManagerAddress = this.originalRequestManagerAddress;
+  }
+  async sendRequestTransaction(provider: IEthereumProvider): Promise<void> {
```

Comment:
Most of the times this happens by accident. Can we enforce this in the linter? 
I actually also prefer adding an empty line after a function declaration.

Reply:
Just did a quick research. There are ESLint rules to achieve this:
https://eslint.org/docs/latest/rules/padded-blocks is probably the way to go.
We could also consider these two:
https://eslint.org/docs/latest/rules/padding-line-between-statements
https://eslint.org/docs/latest/rules/lines-between-class-members

Since applying these will change a lot of code, I would keep it out of the scope of this PR and just create an issue for now.

Reply:
I created one: #2048

Reply:
Exactly, with the right combination and definition of those 3 rules, we can achieve it. Thanks for creating the issue

Reply:
For the time being, i can just separate these functions with spaces without changing other parts of the code.